kmm/positions/wire_camera_positions.py

```diff
@@ -1,47 +1,54 @@
+import numpy as np
 from pydantic import validate_arguments
 
 from kmm import CarDirection
 from kmm.positions.positions import Positions
 
 
 @validate_arguments(config=dict(arbitrary_types_allowed=True))
-def wire_camera_positions(positions: Positions, direction: CarDirection):
-    ascending = kmm_ascending(positions.dataframe)
-
-    if (
-        (direction == CarDirection.A and ascending)
-        or (direction == CarDirection.B and not ascending)
-    ):
-        correction = -8
-
-    elif (
-        (direction == CarDirection.A and not ascending)
-        or (direction == CarDirection.B and ascending)
-    ):
-        correction = 8
+def wire_camera_positions(positions: Positions, car_direction: CarDirection):
 
+    if car_direction == CarDirection.A:
+        corrections = kmm_directions(positions.dataframe) * -8
+    elif car_direction == CarDirection.B:
+        corrections = kmm_directions(positions.dataframe) * 8
     else:
-        raise ValueError
+        raise ValueError(car_direction)
 
     return positions.replace(
         dataframe=(
             positions.dataframe
-            .assign(meter=lambda df: df["meter"] + correction)
+            .assign(meter=lambda df: df["meter"] + corrections)
         )
     )
 
 
-def kmm_ascending(dataframe):
-    total_meter = dataframe["kilometer"] * 1000 + dataframe["meter"]
-    diff = total_meter.values[:-1] - total_meter.values[1:]
-    descending = (diff < 0).mean()
-    ascending = (diff > 0).mean()
+def kmm_directions(df):
 
-    if descending < 0.9 and ascending < 0.9:
-        raise ValueError("Unable to determine ascending/descending kmm numbers")
+    total_meter = (df["kilometer"] * 1000 + df["meter"]).values
```

Comment:
This doesn't work right? We can't count on a "kilometer" being less than 1000 meters?

Reply:
It works because we take the mean but if we wanna be extra cautious we could take into account the >1000m kms. What we do with total meter is just compare the next entry to the current to see if it's ascending/descending and then take the mean, so a >1000m km will just contribute with a small fraction of error. Do you think it's worth taking into account? If so I would just stretch/shrink all meters to be on a scale 0-999 for each km.